src/tempo_config.py

```diff
@@ -3,164 +3,217 @@
 
 """Helper module for interacting with the Tempo configuration."""
 
+import enum
 import logging
 import re
-from dataclasses import asdict
-from typing import Any, Dict, List, Literal, Optional, Union
+from pathlib import Path
+from typing import Any, Dict, List, Optional
 from urllib.parse import urlparse
 
-from pydantic import BaseModel, Field, model_validator, validator
-from pydantic.dataclasses import dataclass as pydantic_dataclass
+from pydantic import BaseModel, ConfigDict, field_validator, model_validator
 
 S3_RELATION_NAME = "s3"
 BUCKET_NAME = "tempo"
 
 logger = logging.getLogger(__name__)
 
 
-class InvalidConfigurationError(Exception):
-    """Invalid configuration."""
+class ClientAuthTypeEnum(str, enum.Enum):
+    """client auth types"""
 
-    pass
+    VERIFY_CLIENT_CERT_IF_GIVEN = "VerifyClientCertIfGiven"
```

Comment:
please add the other possible values:
https://pkg.go.dev/crypto/tls#ClientAuthType

Reply:
(and maybe add this link in a comment to document where we're grabbing these values from)